### PR TITLE
Hash

### PR DESCRIPTION
**Pull request by lfundaro**
hash composed by `Namespace + ServiceName + Host + Path`.

### CODE REVIEW

File: data.go [hunk 1]
```diff
@@ -33,6 +41,7 @@ func ToIngressifyRule(il *v1beta1.IngressList) []IngressifyRule {
 				ir.Path = path.Path
 				ir.ServiceName = path.Backend.ServiceName
 				ir.ServicePort = path.Backend.ServicePort.IntVal
+				ir.Hash = hash(ing.Namespace + path.Backend.ServiceName + ir.Host + ir.Path)
```
Comment (rdsubhas) [inline data.go]: Do you think we should also mixin the ingress name here :)

File: data_test.go [hunk 2]
```diff
@@ -80,6 +85,9 @@ func TestGroupByHost(t *testing.T) {
 			t.Errorf("Missing rule, Name: %s, Namespace: %s, Host: %s, Path: %s, ServicePort: %d, ServiceName: %s",
 				r.Name, r.Namespace, r.Host, r.Path, r.ServicePort, r.ServiceName)
 		}
+		if r.Hash == 0 {
+			t.Errorf("Invalid hash for Ingressify Rule. Hash = %d", r.Hash)
```
Comment (rdsubhas) [inline data_test.go]: Do we need to test this here? Because this seems unrelated to GroupByHost...

File: data_test.go [hunk 3]
```diff
@@ -112,6 +120,9 @@ func TestGroupByPath(t *testing.T) {
...
```
Comment (rdsubhas) [inline data_test.go]: Similarly here, this seems unrelated to GroupByPath

File: data_test.go [hunk 4]
```diff
@@ -49,6 +49,11 @@ func TestToIngressifyRule(t *testing.T) {
 			t.Errorf("Missing rule: %s", missingRule)
 		}
 	}
+	for _, r := range ingressifyRules {
+		if r.Hash == 0 {
+			t.Errorf("Invalid hash for Ingressify Rule. Hash = %d", r.Hash)
```
Comment (rdsubhas) [inline data_test.go]: Just a suggestion, pls feel free to ignore :) But can we test the hash extensively here, and ignore testing it in other places :) i.e. Here we test if the hash really changes if any of the input values change (i.e. if any of namespace/name/service/host/port changes, hash should also change). This can be excessive, hence optional and feel free to ignore :)

**Review by rdsubhas — CHANGES_REQUESTED** [review]

File: data.go (hunk 1 above)
Comment (lfundaro) [reply data.go]: yeah, im putting the `ing.Name` as well.

File: data_test.go (hunk 4 above)
Comment (lfundaro) [reply data_test.go]: I think i would rather skip the hash test. if I want this test to be blackbox style then I would need to test that all possible combinations of variations of `(name,namespace,host,path)` makes the hash to change and at that point it feels I'm testing the hash library itself. I think ideally we would want to be able to test that `name, namespace,... == unhash(hash(namespace,host,path))` , but of course, this is not possible

**Review by rdsubhas — APPROVED** [review]
